File: server/enums/appearance/eye_color.py
from django.db import models


class EyeColor(models.TextChoices):
    """
    Enum for eye color options in image generation.
    """

    BROWN = "brown", "Brown"
    BLUE = "blue", "Blue"
    GREEN = "green", "Green"
    HAZEL = "hazel", "Hazel"
    GRAY = "gray", "Gray"
    AMBER = "amber", "Amber"

    def __str__(self) -> str:
        """
        Get a human-readable string representation of the eye color.
        """
        return self.label

File: client/src/enums/appearance/eyeColor.ts
/**
 * Represents eye color options for image generation.
 * Used in the User model to store appearance preferences.
 */
export enum EyeColor {
  BROWN = "brown",
  BLUE = "blue",
  GREEN = "green",
  HAZEL = "hazel",
  GRAY = "gray",
  AMBER = "amber",
}

/**
 * Maps eye color values to their human-readable display names
 */
export const EYE_COLOR_DISPLAY_NAMES: Record<EyeColor, string> = {
  [EyeColor.BROWN]: "Brown",
  [EyeColor.BLUE]: "Blue",
  [EyeColor.GREEN]: "Green",
  [EyeColor.HAZEL]: "Hazel",
  [EyeColor.GRAY]: "Gray",
  [EyeColor.AMBER]: "Amber",
};
